platforms/core-runtime/process-services/src/main/java/org/gradle/process/internal/ExecHandle.java

```diff
@@ -55,6 +57,12 @@ public interface ExecHandle {
      */
     ExecResult waitForFinish();
 
+    /**
+     * Returns the result of the process execution, if it has finished.  If the process has not finished, returns null.
+     */
+    @Nullable
+    ExecResult getExecResult();
```

Comment:
🤔 I wonder about the need for this method.  This interface already allows for querying the state `getState()` and getting a result if the the process has finished `waitForFinish()`.  

It seems like consumers of `getExecResult()` could check the state and then call `waitForFinish()` if the worker is done, removing the need for this nullable method.

Reply:
It's a good point, and I really didn't want to add `getExecResult`, but the problem is that `waitForFinish()` _always_ throws an exception if it's not successful so you can't get the exec result unless it's successful.  This is old code and used in so many places, changing the behavior of `waitForFinish()` could mushroom into a big change.  I did consider just adding a boolean overload to `waitForFinish()` where the boolean allows you to inhibit the failure.  But it seemed more direct to just allow you to retrieve the result from the handle if you needed it (even if there is a failure).

Reply:
Oh, I see.